metrics/release.go

```diff
@@ -208,6 +236,38 @@ func RegisterCompletedReleaseProcessing(startTime, completionTime *metav1.Time,
 	ReleaseConcurrentProcessingsTotal.WithLabelValues().Dec()
 }
 
+// RegisterValidatedRelease registers a Release as validated, adding a new observation for the
+// Release validated seconds. If either the startTime or the validationTime are nil,
+// no action will be taken.
+func RegisterValidatedRelease(startTime, validationTime *metav1.Time, reason, target string) {
+	if validationTime == nil || startTime == nil {
+		return
+	}
+
+	ReleaseValidationSeconds.
+		With(prometheus.Labels{
+			"reason": reason,
+			"target": target,
+		}).
+		Observe(validationTime.Sub(startTime.Time).Seconds())
+}
+
+// RegisterStartedReleaseProcessing registers a Release as started processing, adding a new observation for the
+// Release start processing duration and increasing the number of concurrent processings. If either the
+// processingStartTime or the releaseStartTime are nil, no action will be taken.
+func RegisterStartedReleaseProcessing(startTime, processingStartTime *metav1.Time, reason, target string) {
```

Comment:
You don't need a new function I think. You should use the `RegisterNewReleaseProcessing` as that's the moment where the phase changes.

Reply:
Function removed. I moved the content of it to the `RegisterNewReleaseProcessing`.